### ISSUE TITLE
OneShot signal firing twice when retriggered in the signal's own callback

### ISSUE THREAD

**Issue by andy-noisyduck** [godotengine/godot#28857]
**Godot version:**
3.1.1 (Mono)

**Issue description:**
Under a rather specific scenario, I am able to get a one-shot signal to fire twice. On the second firing it will trigger the following error (presumably from trying to automatically disconnect the one-shot signal, but it's already been disconnected following the first firing).

    E 0:00:01:0299   Nonexistent signal: resized
      <C Source>     core/object.cpp:1533 @ _disconnect()

This behaviour only happens when using a C# version of the script. A functionally equivalent GDScript does not trigger the error.

**Steps to reproduce:**

1. With a base Control node, add a child HBoxContainer
2. Add a number of children to the container with some size. I used ColorRects. The bug seems to require a minimum of at least 2 children, or it will not fire. 
3. Connect to the "resized" signal of the container, with the one-shot flag set.
4. During the receiver for the resized signal, move the container, *and* set a custom separation. Performing just one of these in isolation is not enough to trigger the bug.
5. The resize event will fire again and trigger the error. 

**Minimal reproduction project:**
[Testing-OneShot.zip](https://github.com/godotengine/godot/files/3174155/Testing-OneShot.zip)
Project includes C# and GDScript versions of the same script to show it's C# only.

**Comment by akien-mga** [comment]
Tested in the current master branch and it now behaves the same with GDScript and C#.
For both scripts, the signal is emitted three times, two run the function as expected and the third triggers an error:
```
Signal fired
Signal fired
ERROR: _disconnect: Nonexistent signal 'resized' in [HBoxContainer:1172].
   At: core/object.cpp:1518.
```

Relabeling as a `core` issue.

I'm not sure this is a bug per se or that it's fixable, but it's worth discussing.

**Comment by KoBeWi** [comment]
Can anyone still reproduce this bug in Godot 3.2.3 or any later release?

If yes, GDScript MRP would be useful.

**Comment by andy-noisyduck** [comment]
> Can anyone still reproduce this bug in Godot 3.2.3 or any later release?
> 
> If yes, GDScript MRP would be useful.

@KoBeWi The original reproduction project zip contains a GDScript version alongside the C# one. They both still trigger the bug for me on 3.2.3 - though it's a rather niche bug.

**Comment by KoBeWi** [comment]
Ah sorry, I didn't notice the .gd file.
This is also still valid in 0167bfa

**Comment by kleonc** [comment]
Fixed by #89451 (since v4.3.dev6.official [89850d553]).
